src/cache_cf.cc

```diff
@@ -4320,10 +4320,8 @@ static void free_icap_service_failure_limit(Adaptation::Icap::Config *cfg)
 #if USE_OPENSSL
 static void parse_sslproxy_cert_adapt(sslproxy_cert_adapt **cert_adapt)
 {
-    char *al;
-    sslproxy_cert_adapt *ca = (sslproxy_cert_adapt *) xcalloc(1, sizeof(sslproxy_cert_adapt));
-    if ((al = ConfigParser::NextToken()) == NULL) {
-        xfree(ca);
+    auto *al = ConfigParser::NextToken();
+    if (al) {
```

Comment:
The above should be written as

```C++
    if (ConfigParser::NextToken()) {
```

... which exposes a PR bug (and the lack of PR testing) -- a missing negation of the `al` condition:

```suggestion
    const auto al = ConfigParser::NextToken();
    if (!al) {
```

Reply:
Adding the missing ```!```.

Disagree on the "should". Using the ```if (ConfigParser::NextToken()) {``` style requires nesting the entire function implementation inside that if-statement. That is bad for long/complex functions like this one.